src/simplewallet/simplewallet.cpp

```diff
@@ -9508,6 +9509,32 @@ bool simple_wallet::print_address(const std::vector<std::string> &args/* = std::
     print_address_sub(m_wallet->get_num_subaddresses(m_current_subaddress_account) - 1);
     m_wallet->device_show_address(m_current_subaddress_account, m_wallet->get_num_subaddresses(m_current_subaddress_account) - 1, boost::none);
   }
+  else if (local_args[0] == "mnew")
+  {
+    local_args.erase(local_args.begin());
+    if (local_args.size() != 1)
+    {
+      fail_msg_writer() << tr("Expected exactly one argument for the amount of new addresses");
+      return true;
+    }
+    uint32_t n;
+    if (!epee::string_tools::get_xtype_from_string(n, local_args[0]))
+    {
+      fail_msg_writer() << tr("failed to parse the amount of new addresses: ") << local_args[0];
+      return true;
+    }
+    if (n > MAX_MNEW_ADDRESSES)
+    {
+      fail_msg_writer() << tr("the amount of new addresses must be lower or equal to ") << MAX_MNEW_ADDRESSES;
+      return true;
+    }
+    for (uint32_t i = 0; i < n; ++i)
+    {
+        m_wallet->add_subaddress(m_current_subaddress_account, tr("(Untitled address)"));
+        print_address_sub(m_wallet->get_num_subaddresses(m_current_subaddress_account) - 1);
+        m_wallet->device_show_address(m_current_subaddress_account, m_wallet->get_num_subaddresses(m_current_subaddress_account) - 1, boost::none);
```

Comment:
Do you know if this is correct? Isn’t that showing a subaddress on hardware wallet? Guess that would not work for 1000 subaddresses.

Reply:
I'm not sure if I understood the question fully. This is the same [code used for the `address new` command](https://github.com/codesoap/monero/blob/4c0ea2df00ccb29f8368c892bda0094dcf1a1547/src/simplewallet/simplewallet.cpp#L9510), so I would guess, that it has no more problems, than typing in `address new` a 1000 times.

Reply:
Oh and in case there is a misunderstanding: 1000 is just the **limit** for how many addresses can be generated in a single `address mnew` call. I would not expect people to generate this many subaddresses often.

Reply:
If you generate an address they get displayed on the hardware wallet. This makes sense for one address, but quickly looping through up to 1000 addresses does not make sense. I would remove that line.

Reply:
I was not aware of this interaction of `monero-wallet-cli` and hardware wallets. Thanks for educating me. I will remove the line.

Reply:
The `address mnew` command still works as expected when I remove the line. I have force pushed the change.

Reply:
ok thanks for testing